src/providers/postgres/qgspostgresconn.h

```diff
@@ -183,6 +183,12 @@ class QgsPostgresConn : public QObject
     Q_OBJECT
 
   public:
+    /* @param shared allow using a shared connection. Should never be
+     *        called from a thread other than the main one. An
+     *        assertion guards against that.
+     *
+     *  @deprecated Use QgsPostgresConnPool
```

Comment:
This code isn't exposed in any public api, so why bother deprecating? I'd just immediately remove it instead...